New post about signing defaults change (#92)
sigstore/sigstore-blog: content/signing-default-rekor-v2.md

+++
title = "Sigstore Signing Is Evolving: Keep Your Verifiers Up to Date"
date = "2026-02-26"
tags = ["sigstore","rekor"]
draft = false
author = "Jussi Kukkonen, Google"
type = "post"
+++

_Sigstore Public Good instance on sigstore.dev is switching to [Rekor v2](https://github.com/sigstore/rekor-tiles) as its default transparency log in May 2026. While existing signatures remain valid, older clients may fail to verify signatures created after this switch.

<!--more-->

> sigstore.dev has been running the new Rekor v2 signature transparency log in parallel with the original Rekor v1 [for a few months now](https://blog.sigstore.dev/rekor-v2-ga/): The services seem to be running well and client support looks good. The next step is using Rekor v2 log entries in new Sigstore signatures by default. This post outlines the plan and addresses compatibility concerns inherent in changing signature content.

Reviewer: ```suggestion
sigstore.dev has been running the new Rekor v2 signature transparency log in parallel with the original Rekor v1 [for a few months now](https://blog.sigstore.dev/rekor-v2-ga/). The services seem to be running well and client support looks good. The next step is using Rekor v2 log entries in new Sigstore signatures by default. This post outlines the plan and addresses compatibility concerns inherent in changing signature content.
```

### What changes?

> As mentioned, the v2 log is already running: we only need to change client configuration to start using it. Doing this configuration in a smart way requires a bit of infrastructure: Signing clients get the service URLs they need via a `SigningConfig` json file, typically delivered automatically via TUF (The Update Framework) -- this is the same mechanism that verifying clients use to get the `TrustedRoot` file with the trusted public key material.

Reviewer: ```suggestion
The v2 log is already running, we only need to change configuration to tell clients to to start using it. Doing this in a smart way requires a bit of infrastructure: Modern signing clients get the service URLs they need via a `SigningConfig` json file, delivered automatically via TUF (The Update Framework) -- this is the same mechanism that verifying clients use to get the `TrustedRoot` file with the trusted public key material.
```

In May 2026 the Rekor v2 URL will be added to the Public Good instance [`SigningConfig`](https://github.com/sigstore/root-signing/blob/main/targets/signing_config.v0.2.json). This will signal Sigstore clients to start creating signature bundles that contain transparency log entries from the Rekor v2 service instead of Rekor v1. Sigstore clients can still use Rekor v1 if they are still working on supporting v2 or if the user explicitly requests v1 entries.

### How Does This Impact You?

Soon, Sigstore signature bundles will contain Rekor v2 log entries. While most current Sigstore clients can handle these new bundles, older clients will fail to verify them. Here’s what you need to do to prepare.

**For Users of Sigstore CLIs (like Cosign)**

> The best thing you can do is keep your client software up to date (see compatibility table below for details). This will ensure your existing workflows continue to work without interruption. If you're using an older client, you'll still be able to verify and sign signatures with Rekor v1 entries, but you won't be able to verify signatures that use Rekor v2.

Reviewer: I think if we use anchors `#####` instead of bolding, we can link (see compatibility table below) to the table.

**For Sigstore Integrators**

Anyone integrating Sigstore verification should ensure they are using an implementation compatible with Rekor v2.

> The integrators (e.g. package ecosystems) that also control the signing process have more control over the transition. You can choose to pin the Rekor version in your signing process to manage the switchover time. However, if you do this, you are responsible for switching to v2 shortly after the Sigstore defaults change.

Reviewer: ```suggestion
Integrators (e.g. package ecosystems) that manage the signing process have more control over the transition. You can choose to pin the Rekor version in your signing process to manage the switchover time. However, if you do this, you are responsible for switching to v2 shortly after the Sigstore defaults change.
```

$ cosign verify-blob <artifact> --bundle=<bundle.sigstore.json> --certificate-identity=<identity> --certificate-oidc-issuer=<issuer>
Verified OK

# for images
$ cosign verify <image> --certificate-identity=<identity> --certificate-oidc-issuer=<issuer>
  ...
```

If you are verifying container images with older cosign clients, you may also encounter issues for signatures that use the [OCI referrers API](https://github.com/sigstore/cosign?tab=readme-ov-file#oci-artifacts). Ensure your clients are up to date to find OCI referrers signatures and handle new Sigstore signatures.

**Why not just hardcode the log URL?**

`SigningConfig` was not always part of the Sigstore design (and as a result support for it is not yet universal, especially in older client releases). It was added because dynamic discovery of service URLs allows clients to support other Sigstore instances, enables adding entries in multiple logs, and allows instance maintainers to rotate transparency log instances without service disruption. Log rotation was always planned but has been avoided so far because it would be disruptive, as the log URL is hardcoded in many places.

**Which clients exactly are compatible?**

> This is an overview of the situation at time of writing: please refer to [Sigstore Conformance](https://github.com/sigstore/sigstore-conformance/) client Conformance Report for up-to-date conformance test results and the client projects documentation for details.

Reviewer: should we link to the report directly?

Author: It currently has the flaw of not linking back to conformance README so I think not

+++
title = "Signing defaults are changing: Rekor v2 used as transparency log"
date = "2026-03-03"
tags = ["sigstore","rekor"]
draft = true
author = "Jussi Kukkonen, Google"
type = "post"
+++

> The Sigstore Public Good instance on sigstore.dev has been running the new Rekor v2 signature transparency log in parallel with the original Rekor v1 for a few months now: The services seem to be running well and client support looks good. The next step is using Rekor v2 log entries in new Sigstore signatures by default. This post has details the plans and addresses compatibility concerns inherent in changing signature content.

Reviewer 2: You can link to https://blog.sigstore.dev/rekor-v2-ga/ in this first section as a primer.

### What changes?

As mentioned, the v2 log is already running: we only need to change client configuration to start using it. Doing this configuration in a smart way requires a bit of infrastructure: Signing clients get the service URLs they need via a `SigningConfig` json file, typically delivered automatically via TUF (The Update Framework) -- this is the same mechanism that verifying clients use to get the `TrustedRoot`  file with the trusted public key material.

> By end of March 2026 the Rekor v2 URL will be added to the Public Good instance `SigningConfig`: this is a signal to Sigstore clients to start creating signature bundles that contain transparency log entries from the Rekor v2 service instead of Rekor v1. Sigstore clients can still keep using Rekor v1 if they are still working on supporting v2 or if the user explicitly requests v1 entries.

Reviewer 2: > if the user explicitly requests v1 entries

Do any of our clients allow this? Or do we mean for those building on our SDKs?

Reviewer: sigstore-java:2.0.0 requires explicitly enabling rekor v2. I should probably get in there and remove that.

Author: I was mostly thinking SDK users but I think the line blurs quite fast if e.g. a package ecosystem uses SDK or CLI in specific way.

Author: > Do any of our clients allow this?

Python does, so that an override exists if some use case really really was not ready for v2

> By end of March 2026 the Rekor v2 URL will be added to the Public Good instance `SigningConfig`: this is a signal to Sigstore clients to start creating signature bundles that contain transparency log entries from the Rekor v2 service instead of Rekor v1. Sigstore clients can still keep using Rekor v1 if they are still working on supporting v2 or if the user explicitly requests v1 entries.

Reviewer 2: Can we mention the date up at the top too, as a 1-line summary of this post?

### How does this impact users?

Soon there will be Sigstore signature bundles out there containing Rekor v2 log entries. Current releases of most Sigstore clients will verify these new signature bundles without issues, but older clients will be unable to verify them.

**Advice to users of Sigstore CLIs such as cosign**

The good news is that standard best practices will just work: Keep your Sigstore client software up to date (see end of post for specific client version information) and you should not notice a thing: existing workflows will keep working.

**Advice to Sigstore integrators**

> Sigstore integrators (like package ecosystems) who control the signing process but do not necessarily control the verifying software may want to consider forcing rekor version 1 when signing _for a limited time_: this gives verifiers more time to upgrade their software.

Reviewer 2: We should note that the releases that support verifying rekor v2 entries have been out for a bit. Cosign not being updated is less of a concern since that won't affect any package ecosystems.

I'm hesitant to suggest doing anything for a limited time since that risks not actually being limited. Maybe we say reach out if you have concerns?

Reviewer 3: > I'm hesitant to suggest doing anything for a limited time since that risks not actually being limited.

I agree with this, there are some ecosystem where support is needed for a time yet known (we need to figure out when all supported npm versions supports rekor v2 as an example).

I wonder if this post should also state that integrators should prioritize _verification_ now for rekor v2 entries?

Author: I will rephrase: I'm mostly thinking of ecosystems that are using updated signing tools but don't want to rely on major changes happening at a surprising time (think https://github.com/pypa/gh-action-pypi-publish switching to rekor2) I think it's reasonable for them to control the time of that change -- otherwise we're rewarding the ecosystems that lag behind (as they won't have this surprise situation) and punishing the ones that are up-to-date.

> Integrators should still prepare for more frequent log rotation in future: make sure your signing process does not hard code service URLs and instead uses SigningConfig (either via TUF or a more DIY mechanism)

Reviewer 2: +1, let's bold this. This is different than rekor.sigstore.dev where we know it's baked into pipelines.

### FAQ

**Why not just hard code the log URL?**

> SigningConfig was not always part of the Sigstore design (and as a result support for it is not yet universal, especially in older client releases). It was added because dynamically discovering service URLs was found to be _very_ useful: it not only makes clients ready to support other Sigstore instances but also allows instance maintainers to rotate transparency log instances without service disruption. Log rotation was always in the plans but has been avoided so far because of how disruptive it would have been -- because the log URL has been hard coded in so many places.

Reviewer 2: Might also be worth mentioning it enables discovery of other logs too for when a signer wants to log to multiple.

### How Does This Impact You?

Soon, Sigstore signature bundles will start containing Rekor v2 log entries. While most current Sigstore clients can handle these new bundles, older clients will fail to verify them. Here’s what you need to do to prepare.

**For Users of Sigstore CLIs (like Cosign)**

The best thing you can do is keep your client software up to date (see compatibility table below for details). This will ensure your existing workflows continue to work without interruption. If you're using an older client, you'll still be able to verify and sign signatures with Rekor v1 entries, but you won't be able to verify signatures that use Rekor v2.

**For Sigstore Integrators**

As an integrator (e.g., a package ecosystem), you have more control over the transition. You can choose to pin the Rekor version in your signing process to manage the switchover time. However, if you do this, you are responsible for switching to v2 shortly after the Sigstore defaults change.

> [!NOTE] Integrators must still prepare for more frequent log rotation in future: make sure your signing process does not hard code service URLs and instead uses SigningConfig (either via TUF or a more DIY mechanism)

Author: no support for "[!NOTE]", need to change this

### FAQ

**Why not just hard code the log URL?**

SigningConfig was not always part of the Sigstore design (and as a result support for it is not yet universal, especially in older client releases). It was added because dynamically discovering service URLs was found to be _very_ useful: it makes clients ready to support other Sigstore instances, enables adding entries in multiple logs, an allows instance maintainers to rotate transparency log instances without service disruption. Log rotation was always in the plans but has been avoided so far because of how disruptive it would have been -- because the log URL has been hard coded in so many places.

**Which clients exactly are compatible?**

This is an overview of the situation at time of writing: please refer to [Sigstore Conformance](https://github.com/sigstore/sigstore-conformance/) client Conformance Report for up-to-date conformance test results and the client projects documentation for details.

> | Compatible client or library | First version with Rekor v2 support |

Author: there's no CSS at all for tables so this looks awful

+++
title = "Signing defaults are changing: Rekor v2 used as transparency log"
date = "2026-02-26"
tags = ["sigstore","rekor"]
draft = false
author = "Jussi Kukkonen, Google"
type = "post"
+++

> _Sigstore Public Good instance on sigstore.dev is switching to Rekor v2 as the default signature transparency log before end of March 2026._

Reviewer 2: ```suggestion
_Sigstore Public Good instance on sigstore.dev is switching to [Rekor v2](https://github.com/sigstore/rekor-tiles) as the default signature transparency log before end of March 2026._
```

<!--more-->

sigstore.dev has been running the new Rekor v2 signature transparency log in parallel with the original Rekor v1 [for a few months now](https://blog.sigstore.dev/rekor-v2-ga/): The services seem to be running well and client support looks good. The next step is using Rekor v2 log entries in new Sigstore signatures by default. This post outlines the plan and addresses compatibility concerns inherent in changing signature content.

### What changes?

As mentioned, the v2 log is already running: we only need to change client configuration to start using it. Doing this configuration in a smart way requires a bit of infrastructure: Signing clients get the service URLs they need via a `SigningConfig` json file, typically delivered automatically via TUF (The Update Framework) -- this is the same mechanism that verifying clients use to get the `TrustedRoot` file with the trusted public key material.

> By end of March 2026 the Rekor v2 URL will be added to the Public Good instance `SigningConfig`: this is a signal to Sigstore clients to start creating signature bundles that contain transparency log entries from the Rekor v2 service instead of Rekor v1. Sigstore clients can still keep using Rekor v1 if they are still working on supporting v2 or if the user explicitly requests v1 entries.

Reviewer 2: can we link to the PGI signing config in case anyone's curious?

Author: sure. I linked to the file in root-signing, I think that makes most sense
